### PR TITLE
omnibus: add an autotools wrapper

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR centralizes all of our custom software rules calls to `./configure` / `make [install]` into a single location

### Motivation

A lot of boilerplate is duplicated and sometimes invalid (passing `-I` as `LDFLAGS` or `-L` as `CFLAGS`)
This also has the advantage of allowing a single place for tweaking builds of autotools based packages

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by KevinFairise2 — COMMENTED** [review]
That's nice!

File: omnibus/config/software/librdkafka.rb [hunk 1]
```diff
@@ -18,15 +20,15 @@
...
-    "CFLAGS" => "-L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include",
-    "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
-  }
-
-  command "./configure --enable-sasl --prefix=#{install_dir}/embedded", :env => env
-  command "make -j #{workers}", :env => env
-  command "make install", :env => env
+  build_with_autotools({
+    :discard_default_opts => true,
+    :prefix => "#{install_dir}/embedded",
```
[the remark this answers is not in the document]
Comment (chouquette) [reply omnibus/config/software/librdkafka.rb:25]: You are correct, thanks for noticing!

File: omnibus/config/software/cyrus-sasl.rb [hunk 2]
```diff
@@ -17,23 +19,12 @@
   license "Carnegie Mellon University license"
   license_file "https://raw.githubusercontent.com/cyrusimap/cyrus-sasl/master/COPYING"
 
-  env = {
-    "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib",
-    "CFLAGS" => "-L#{install_dir}/embedded/lib -I#{install_dir}/embedded/include",
-    "LD_RUN_PATH" => "#{install_dir}/embedded/lib",
-  }
```
[the remark this answers is not in the document]
Comment (chouquette) [reply omnibus/config/software/cyrus-sasl.rb:24]: > If I understand correctly we now use with_standard_compiler_flags to provide these env variables.

Indeed

> but on other platform with_standard_build_flags returns other flags. Could it cause us issues?

As far as  I can see, no; All platforms provide identical or very similar environments values for `CFLAGS` / `LDFLAGS`. 
The main concern I had was the lack of `rpath` for macOS, but this is handled/worked around by also providing `LD_RUN_PATH` with the same value, which will kick in in the absence of an `rpath` value

Omnibus' healthcheck should also catch any issue related to linking with a library outside of the install dir.

**Comment by chouquette** [comment]
Closing as this was done differently in #20500

**Pull request closed without merging**